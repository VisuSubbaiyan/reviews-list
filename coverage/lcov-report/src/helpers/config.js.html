<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/helpers/config.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/helpers</a> config.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/1</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20</td><td class="line-coverage quiet"><span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const theme = {
  primary: '#20232a',
  secondary: '#ccc',
  black: '#282c34',
  mediumGrey: '#aaa',
  lightGrey: '#dcdcdc',
  mediumFont: '14px',
  smallFont: '10px',
  linkColor: '#0366d6',
  blue: '#39a2d6',
  blueGrey: '#7998a8',
  formControlRadius: '5px',
  boxRadius: '10px',
  buttonRadius: '15px',
  breakpoints: {
    small: 640,
    medium: 769
  }
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon May 27 2019 19:38:35 GMT+0200 (CEST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
