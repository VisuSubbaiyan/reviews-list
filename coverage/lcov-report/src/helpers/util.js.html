<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/helpers/util.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/helpers</a> util.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">53.13% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>17/32</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.18% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">46.15% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/26</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">40x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { css } from 'styled-components';
import { format as formatDate, endOfWeek, startOfWeek, startOfMonth } from 'date-fns';
&nbsp;
/** Styled compoenent uitls */
// Based on the number of codes we can move this to seperate file
export const styling = props =&gt; props.css &amp;&amp; css`${props.css}`;
&nbsp;
export const viewport = breakpoint =&gt; (...args) =&gt; css`
    @media (min-width: ${breakpoint}px) {
      ${css(...args)}
    }
  `;
&nbsp;
export const iconSizes = {
  xs: '12px',
  s: '16px',
  m: '24px',
  l: '32px',
  xl: '38px',
};
&nbsp;
export const getSize = props =&gt; iconSizes[props.size];
&nbsp;
&nbsp;
/** Reviews component utils */
export const formatTitle = <span class="fstat-no" title="function not covered" >(t</span>itle, group) =&gt; {
<span class="cstat-no" title="statement not covered" >  switch (group) {</span>
    case 'week':
<span class="cstat-no" title="statement not covered" >      return `${formatDate(title, 'DD.MM')} - ${formatDate(endOfWeek(title), 'DD.MM.YYYY')}`;</span>
    case 'month':
<span class="cstat-no" title="statement not covered" >      return formatDate(title, 'MMMM YYYY');</span>
    default:
<span class="cstat-no" title="statement not covered" >      return formatDate(title, 'DD MMMM YYYY');</span>
  }
};
&nbsp;
export const isTextMatch = <span class="fstat-no" title="function not covered" >(t</span>ext, match) =&gt; {
  const textLowerCase = <span class="cstat-no" title="statement not covered" >text.toLowerCase();</span>
  const matchLowerCase = <span class="cstat-no" title="statement not covered" >match.toLowerCase();</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >  return textLowerCase.includes(matchLowerCase);</span>
};
&nbsp;
export const isReviewMatch = <span class="fstat-no" title="function not covered" >(r</span>eview, matchWith) =&gt; {
  const { title, content } = <span class="cstat-no" title="statement not covered" >review;</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >  return matchWith &amp;&amp; (isTextMatch(title, matchWith) || isTextMatch(content, matchWith));</span>
};
&nbsp;
export const groupByIterator = group =&gt; <span class="fstat-no" title="function not covered" >re</span>view =&gt; {
  // GroupBy
  // Day new Date();
  // Week startOfWeek(new Date());
  // Month startOfMonth(new Date());
  const reviewedDate = <span class="cstat-no" title="statement not covered" >new Date(review.reviewCreated);</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >  switch (group) {</span>
    case 'week': {
<span class="cstat-no" title="statement not covered" >      return startOfWeek(reviewedDate);</span>
    }
    case 'month': {
<span class="cstat-no" title="statement not covered" >      return startOfMonth(reviewedDate);</span>
    }
    default: {
<span class="cstat-no" title="statement not covered" >      return reviewedDate;</span>
    }
  }
}
&nbsp;
export const sortByOptions = [{ value: 'desc', name: 'Most Recent' }, { value: 'asc', name: 'Older' }];
&nbsp;
export const groupByOptions = [
  { value: 'day', name: 'Group by day' },
  { value: 'week', name: 'Group by week' },
  { value: 'month', name: 'Group by month' }
];
&nbsp;
export const filterOptions = [
  { value: 5, name: '5 star only' },
  { value: 4, name: '4 star only' },
  { value: 3, name: '3 star only' },
  { value: 2, name: '2 star only' },
  { value: 1, name: '1 star only' }
];
&nbsp;
export const smoothScrollToPosition = <span class="fstat-no" title="function not covered" >to</span>p =&gt; <span class="cstat-no" title="statement not covered" >window.scrollTo({</span>
    top,
    behavior: 'smooth',
  });
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon May 27 2019 19:38:35 GMT+0200 (CEST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
