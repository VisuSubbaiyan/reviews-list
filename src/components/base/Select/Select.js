import React, { Component } from 'react';
import PropTypes from 'prop-types';
import StyledSelect from './Select.style';

/**
 * `Select` - HTML select (need to modify the options custom options )
 */
class Select extends Component {
  static defaultProps = {
    selected: '',
    options: [{ value: '1', name: '1' }, { value: '2', name: '2' }],
    disabled: false
  }

  /**
   * @property {boolean} selected - current selected value
   * @property {Array<Object>} options - options to be rendered in select box
   * @property {boolean} disabled - idicate that select box is disabled or not
   * @property {function} onChange - handle select options
   */
  static propsTypes = {
    selected: PropTypes.any,
    options: PropTypes.arrayOf(PropTypes.object),
    disabled: PropTypes.bool,
    onChange: PropTypes.func
  }

  state = {
    selected: this.props.selected,
    disabled: this.props.disabled
  }

  componentWillReceiveProps({ disabled }) {
    if (this.props.disabled !== disabled) {
      this.setState({ disabled });
    }
  }

  handleOnChange = event => {
    this.setState({ selected: event.target.value });
    this.props.onChange(event.target.value, event); // Passing event object also for future reference 
  }

  render() {
    return (
      <StyledSelect
        onChange={this.handleOnChange}
        selected={this.state.selected}
        disabled={this.state.disabled}
        css={this.props.css}
      >
        {this.props.options.map(({ value, name }) => <option value={value} key={name}>{name}</option>)}
      </StyledSelect>
    );
  }
}

export default Select;
