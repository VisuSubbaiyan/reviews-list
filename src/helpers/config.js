export const theme = {
  primary: '#20232a',
  secondary: '#ccc',
  black: '#282c34',
  mediumGrey: '#aaa',
  lightGrey: '#dcdcdc',
  mediumFont: '14px',
  smallFont: '10px',
  linkColor: '#0366d6',
  blue: '#39a2d6',
  blueGrey: '#7998a8',
  formControlRadius: '5px',
  boxRadius: '10px',
  buttonRadius: '15px',
  breakpoints: {
    small: 640,
    medium: 769
  }
};
