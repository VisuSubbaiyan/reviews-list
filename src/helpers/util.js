import { css } from 'styled-components';
import { format as formatDate, endOfWeek, startOfWeek, startOfMonth } from 'date-fns';

/** Styled compoenent uitls */
// Based on the number of codes we can move this to seperate file
export const styling = props => props.css && css`${props.css}`;

export const viewport = breakpoint => (...args) => css`
    @media (min-width: ${breakpoint}px) {
      ${css(...args)}
    }
  `;

export const iconSizes = {
  xs: '12px',
  s: '16px',
  m: '24px',
  l: '32px',
  xl: '38px',
};

export const getSize = props => iconSizes[props.size];


/** Reviews component utils */
export const formatTitle = (title, group) => {
  switch (group) {
    case 'week':
      return `${formatDate(title, 'DD.MM')} - ${formatDate(endOfWeek(title), 'DD.MM.YYYY')}`;
    case 'month':
      return formatDate(title, 'MMMM YYYY');
    default:
      return formatDate(title, 'DD MMMM YYYY');
  }
};

export const isTextMatch = (text, match) => {
  const textLowerCase = text.toLowerCase();
  const matchLowerCase = match.toLowerCase();

  return textLowerCase.includes(matchLowerCase);
};

export const isReviewMatch = (review, matchWith) => {
  const { title, content } = review;

  return matchWith && (isTextMatch(title, matchWith) || isTextMatch(content, matchWith));
};

export const groupByIterator = group => review => {
  // GroupBy
  // Day new Date();
  // Week startOfWeek(new Date());
  // Month startOfMonth(new Date());
  const reviewedDate = new Date(review.reviewCreated);

  switch (group) {
    case 'week': {
      return startOfWeek(reviewedDate);
    }
    case 'month': {
      return startOfMonth(reviewedDate);
    }
    default: {
      return reviewedDate;
    }
  }
}

export const sortByOptions = [{ value: 'desc', name: 'Most Recent' }, { value: 'asc', name: 'Older' }];

export const groupByOptions = [
  { value: 'day', name: 'Group by day' },
  { value: 'week', name: 'Group by week' },
  { value: 'month', name: 'Group by month' }
];

export const filterOptions = [
  { value: 5, name: '5 star only' },
  { value: 4, name: '4 star only' },
  { value: 3, name: '3 star only' },
  { value: 2, name: '2 star only' },
  { value: 1, name: '1 star only' }
];

export const smoothScrollToPosition = top => window.scrollTo({
    top,
    behavior: 'smooth',
  });
