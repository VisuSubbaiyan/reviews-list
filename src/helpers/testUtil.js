import React from 'react';
import { mount, shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { prettyPrint } from 'html';

import { theme } from './config';
import { UPDATE_LOADING } from '../redux/actions';

export const createMountedComponent = ({ children, isShallow }) => {
  const component = (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  );
  const mountedComponent = isShallow ? shallow(component) : mount(component);
  mountedComponent.htmlRender = () => prettyPrint(mountedComponent.html(), { indent_size: 2 });

  return mountedComponent;
};

export const sampleText = 'sample text';
export const mockFunction = jest.fn(value => value);

export const inputProps = {
  value: 'sample name',
  onChange: mockFunction,
  placeholder: 'sample placeholder'
};

export const sampleState = {
  search: '',
  stars: 5,
  groupBy: 'day',
  sortBy: 'desc',
  diableSort: false
};

export const updateLoadingAction = {
  type: UPDATE_LOADING,
  payload: { loading: true }
};

export const expectedLoadingState = {
  page: 1,
  loading: true,
  reviews: [],
  hasMore: false
}
