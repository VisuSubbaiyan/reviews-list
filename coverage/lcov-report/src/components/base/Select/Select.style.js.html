<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/base/Select/Select.style.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/components/base/Select</a> Select.style.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styled from 'styled-components';
import { styling } from '../../../helpers/util';
&nbsp;
const StyledSelect = styled.select`
  border-radius: ${props =&gt; props.theme.formControlRadius};
  border: 1px solid ${props =&gt; props.disabled ? <span class="branch-0 cbranch-no" title="branch not covered" >props.theme.secondary </span>: props.theme.blue};
  background-color: ${props =&gt; props.theme.lightGrey};
  font-size: ${props =&gt; props.theme.mediumFont};
  height: 32px;
  width: 100%;
&nbsp;
  ${styling}
`;
&nbsp;
export default StyledSelect;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon May 27 2019 19:38:35 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
