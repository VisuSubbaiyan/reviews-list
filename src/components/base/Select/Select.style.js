import styled from 'styled-components';
import { styling } from '../../../helpers/util';

const StyledSelect = styled.select`
  border-radius: ${props => props.theme.formControlRadius};
  border: 1px solid ${props => props.disabled ? props.theme.secondary : props.theme.blue};
  background-color: ${props => props.theme.lightGrey};
  font-size: ${props => props.theme.mediumFont};
  height: 32px;
  width: 100%;

  ${styling}
`;

export default StyledSelect;
