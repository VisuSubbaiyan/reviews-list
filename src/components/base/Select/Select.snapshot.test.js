import React from 'react';
import 'jest-styled-components';

import { createMountedComponent } from '../../../helpers/testUtil';
import Select from './Select';

describe('Select component', () => {
  let mountedSelect;

  beforeEach(() => {
    mountedSelect = createMountedComponent({
      children: <Select onChange={() => null} />
    });
  });

  it('render correctly', () => {
    expect(mountedSelect.htmlRender()).toMatchSnapshot();
    expect(mountedSelect.find('select')).toHaveLength(1);
    expect(mountedSelect.find('option')).toHaveLength(2); // default options
  });

  // it('render disabled component', () => {

  // });

  it('simulate change action', () => {
    mountedSelect.find('select').simulate('change', { target: { value: '2' } });
    expect(mountedSelect.find('select').props().selected).toEqual('2');
  });
})