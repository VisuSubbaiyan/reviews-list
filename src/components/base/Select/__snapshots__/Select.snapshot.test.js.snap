// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component render correctly 1`] = `
"<select class=\\"sc-bdVaJa rCKJb\\">
  <option value=\\"1\\">1</option>
  <option value=\\"2\\">2</option>
</select>"
`;
